pkg/clients/eventstreamsadminv1/eventstreamsadminv1.go

```diff
@@ -0,0 +1,234 @@
+	o.Configs = GenerateConfigUpdate(in.Configs)
+	return nil
+}
+
+// GenerateConfigUpdate generates []arv1.ConfigUpdate from []v1alpha1.ConfigCreate
+func GenerateConfigUpdate(in []v1alpha1.ConfigCreate) []arv1.ConfigUpdate {
+	if in == nil {
+		return nil
+	}
+	o := []arv1.ConfigUpdate{}
+	for _, m := range in {
+		item := arv1.ConfigUpdate{
+			Name:  reference.ToPtrValue(m.Name),
+			Value: reference.ToPtrValue(m.Value),
+			//ResetToDefault: needs to be a pointer to a boolean value, should be true or false??
+		}
+		o = append(o, item)
+	}
+	return o
+}
+
+// GenerateObservation produces TopicObservation object from *arv1.TopicDetail object.
+func GenerateObservation(in *arv1.TopicDetail) (v1alpha1.TopicObservation, error) {
+
+	o := v1alpha1.TopicObservation{
+		ReplicationFactor:  ibmc.Int64Value(in.ReplicationFactor),
+		RetentionMs:        ibmc.Int64Value(in.RetentionMs),
+		CleanupPolicy:      reference.FromPtrValue(in.CleanupPolicy),
+		Configs:            Generatev1alpha1TopicConfigs(in.Configs),
+		ReplicaAssignments: GenerateReplicaAssignments(in.ReplicaAssignments),
+	}
+	return o, nil
+}
+
+// Generatev1alpha1TopicConfigs generates *v1alpha1.TopicConfigs from *arv1.TopicConfigs
+func Generatev1alpha1TopicConfigs(in *arv1.TopicConfigs) *v1alpha1.TopicConfigs {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.TopicConfigs{
+		CleanupPolicy:     reference.FromPtrValue(in.CleanupPolicy),
+		MinInsyncReplicas: reference.FromPtrValue(in.MinInsyncReplicas),
+		RetentionBytes:    reference.FromPtrValue(in.RetentionBytes),
+		RetentionMs:       reference.FromPtrValue(in.RetentionMs),
+		SegmentBytes:      reference.FromPtrValue(in.SegmentBytes),
+		SegmentIndexBytes: reference.FromPtrValue(in.SegmentIndexBytes),
+		SegmentMs:         reference.FromPtrValue(in.SegmentMs),
+	}
+	return o
+}
+
+// GenerateReplicaAssignments generates []v1alpha1.ReplicaAssignment from []arv1.ReplicaAssignment
+func GenerateReplicaAssignments(in []arv1.ReplicaAssignment) []v1alpha1.ReplicaAssignment {
+	if in == nil {
+		return nil
+	}
+	o := []v1alpha1.ReplicaAssignment{}
+	for _, m := range in {
+		item := v1alpha1.ReplicaAssignment{
+			ID:      ibmc.Int64Value(m.ID),
+			Brokers: GenerateBrokers(m.Brokers),
+		}
+		o = append(o, item)
+	}
+	return o
+}
+
+// GenerateBrokers generates *v1alpha1.ReplicaAssignmentBrokers from *arv1.ReplicaAssignmentBrokers
+func GenerateBrokers(in *arv1.ReplicaAssignmentBrokers) *v1alpha1.ReplicaAssignmentBrokers {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.ReplicaAssignmentBrokers{
+		Replicas: in.Replicas,
+	}
+	return o
+}
+
+// IsUpToDate checks whether current state is up-to-date compared to the given set of parameters.
+func IsUpToDate(in *v1alpha1.TopicParameters, observed *arv1.TopicDetail) (bool, error) {
+	desired := in.DeepCopy()
+	actual, err := GenerateTopicParameters(observed)
+	if err != nil {
+		return false, err
+	}
+
+	// it says logger isn't found even when logger is imported??
```

Comment:
You need to import `"github.com/crossplane/crossplane-runtime/pkg/logging"` and also to add  `l logging.Logger` in the fields for IsUpToDate():
```
func IsUpToDate(in *v1alpha1.TopicParameters, observed *arv1.TopicDetail, l logging.Logger) (bool, error) 
```